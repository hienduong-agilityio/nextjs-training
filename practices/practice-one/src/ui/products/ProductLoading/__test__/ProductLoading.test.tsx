import { render, screen } from '@testing-library/react';

// UI
import {
  LoadingProductCard,
  LoadingProductTabs,
  LoadingRelatedProducts,
} from '@/ui';

describe('Loading Components', () => {
  it('renders LoadingProductCard and matches snapshot', () => {
    const { asFragment } = render(<LoadingProductCard />);
    expect(screen.getByLabelText('Loading Product Card')).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders LoadingProductTabs and matches snapshot', () => {
    const { asFragment } = render(<LoadingProductTabs />);
    expect(screen.getByLabelText('Loading Product Tabs')).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders LoadingRelatedProducts with four LoadingProductCard components and matches snapshot', () => {
    const { asFragment } = render(<LoadingRelatedProducts />);
    expect(
      screen.getByLabelText('Loading Related Products'),
    ).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
