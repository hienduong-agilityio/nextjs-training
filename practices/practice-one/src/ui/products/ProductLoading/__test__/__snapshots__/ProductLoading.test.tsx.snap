// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Components renders LoadingProductCard and matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="Loading Product Card"
    class="relative flex flex-col transition-shadow border-2 rounded-sm border-gray-200 animate-pulse"
  >
    <div
      class="absolute top-2 left-2 w-24 h-6 bg-gray-300 rounded-md"
    />
    <div
      class="relative bg-gray-200 h-[273px] w-full flex items-center justify-center rounded-sm"
    >
      <div
        class="w-full h-[200px] sm:h-[273px] bg-gray-300 rounded-md"
      />
    </div>
    <div
      class="absolute inset-0 z-10 flex items-center justify-center gap-2 mx-3 my-5 bg-white opacity-0 bg-opacity-95 group-hover:opacity-100"
    >
      <div
        class="w-12 h-12 bg-gray-300 rounded-full"
      />
      <div
        class="w-12 h-12 bg-gray-300 rounded-full"
      />
    </div>
    <div
      class="flex flex-col items-center justify-between h-full px-4 py-4 bg-white"
    >
      <div
        class="w-full h-6 bg-gray-300 rounded-md mb-1"
      />
      <div
        class="w-1/3 h-4 bg-gray-300 rounded-md mb-2"
      />
      <div
        class="flex items-center gap-2 w-full justify-center"
      >
        <div
          class="w-1/3 h-5 bg-gray-300 rounded-md"
        />
        <div
          class="w-1/4 h-4 bg-gray-200 rounded-md"
        />
        <div
          class="w-1/4 h-4 bg-gray-200 rounded-md"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Loading Components renders LoadingProductTabs and matches snapshot 1`] = `
<DocumentFragment>
  <section
    aria-label="Loading Product Tabs"
    class="mt-8 px-4 py-8 bg-gray-200 rounded-md animate-pulse"
  >
    <div
      class="w-1/3 h-8 bg-gray-300 rounded-md mb-6"
    />
    <div
      class="h-6 w-full bg-gray-300 rounded-md"
    />
  </section>
</DocumentFragment>
`;

exports[`Loading Components renders LoadingRelatedProducts with four LoadingProductCard components and matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="Loading Related Products"
    class="grid w-full grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-6"
  >
    <div
      aria-label="Loading Product Card"
      class="relative flex flex-col transition-shadow border-2 rounded-sm border-gray-200 animate-pulse"
    >
      <div
        class="absolute top-2 left-2 w-24 h-6 bg-gray-300 rounded-md"
      />
      <div
        class="relative bg-gray-200 h-[273px] w-full flex items-center justify-center rounded-sm"
      >
        <div
          class="w-full h-[200px] sm:h-[273px] bg-gray-300 rounded-md"
        />
      </div>
      <div
        class="absolute inset-0 z-10 flex items-center justify-center gap-2 mx-3 my-5 bg-white opacity-0 bg-opacity-95 group-hover:opacity-100"
      >
        <div
          class="w-12 h-12 bg-gray-300 rounded-full"
        />
        <div
          class="w-12 h-12 bg-gray-300 rounded-full"
        />
      </div>
      <div
        class="flex flex-col items-center justify-between h-full px-4 py-4 bg-white"
      >
        <div
          class="w-full h-6 bg-gray-300 rounded-md mb-1"
        />
        <div
          class="w-1/3 h-4 bg-gray-300 rounded-md mb-2"
        />
        <div
          class="flex items-center gap-2 w-full justify-center"
        >
          <div
            class="w-1/3 h-5 bg-gray-300 rounded-md"
          />
          <div
            class="w-1/4 h-4 bg-gray-200 rounded-md"
          />
          <div
            class="w-1/4 h-4 bg-gray-200 rounded-md"
          />
        </div>
      </div>
    </div>
    <div
      aria-label="Loading Product Card"
      class="relative flex flex-col transition-shadow border-2 rounded-sm border-gray-200 animate-pulse"
    >
      <div
        class="absolute top-2 left-2 w-24 h-6 bg-gray-300 rounded-md"
      />
      <div
        class="relative bg-gray-200 h-[273px] w-full flex items-center justify-center rounded-sm"
      >
        <div
          class="w-full h-[200px] sm:h-[273px] bg-gray-300 rounded-md"
        />
      </div>
      <div
        class="absolute inset-0 z-10 flex items-center justify-center gap-2 mx-3 my-5 bg-white opacity-0 bg-opacity-95 group-hover:opacity-100"
      >
        <div
          class="w-12 h-12 bg-gray-300 rounded-full"
        />
        <div
          class="w-12 h-12 bg-gray-300 rounded-full"
        />
      </div>
      <div
        class="flex flex-col items-center justify-between h-full px-4 py-4 bg-white"
      >
        <div
          class="w-full h-6 bg-gray-300 rounded-md mb-1"
        />
        <div
          class="w-1/3 h-4 bg-gray-300 rounded-md mb-2"
        />
        <div
          class="flex items-center gap-2 w-full justify-center"
        >
          <div
            class="w-1/3 h-5 bg-gray-300 rounded-md"
          />
          <div
            class="w-1/4 h-4 bg-gray-200 rounded-md"
          />
          <div
            class="w-1/4 h-4 bg-gray-200 rounded-md"
          />
        </div>
      </div>
    </div>
    <div
      aria-label="Loading Product Card"
      class="relative flex flex-col transition-shadow border-2 rounded-sm border-gray-200 animate-pulse"
    >
      <div
        class="absolute top-2 left-2 w-24 h-6 bg-gray-300 rounded-md"
      />
      <div
        class="relative bg-gray-200 h-[273px] w-full flex items-center justify-center rounded-sm"
      >
        <div
          class="w-full h-[200px] sm:h-[273px] bg-gray-300 rounded-md"
        />
      </div>
      <div
        class="absolute inset-0 z-10 flex items-center justify-center gap-2 mx-3 my-5 bg-white opacity-0 bg-opacity-95 group-hover:opacity-100"
      >
        <div
          class="w-12 h-12 bg-gray-300 rounded-full"
        />
        <div
          class="w-12 h-12 bg-gray-300 rounded-full"
        />
      </div>
      <div
        class="flex flex-col items-center justify-between h-full px-4 py-4 bg-white"
      >
        <div
          class="w-full h-6 bg-gray-300 rounded-md mb-1"
        />
        <div
          class="w-1/3 h-4 bg-gray-300 rounded-md mb-2"
        />
        <div
          class="flex items-center gap-2 w-full justify-center"
        >
          <div
            class="w-1/3 h-5 bg-gray-300 rounded-md"
          />
          <div
            class="w-1/4 h-4 bg-gray-200 rounded-md"
          />
          <div
            class="w-1/4 h-4 bg-gray-200 rounded-md"
          />
        </div>
      </div>
    </div>
    <div
      aria-label="Loading Product Card"
      class="relative flex flex-col transition-shadow border-2 rounded-sm border-gray-200 animate-pulse"
    >
      <div
        class="absolute top-2 left-2 w-24 h-6 bg-gray-300 rounded-md"
      />
      <div
        class="relative bg-gray-200 h-[273px] w-full flex items-center justify-center rounded-sm"
      >
        <div
          class="w-full h-[200px] sm:h-[273px] bg-gray-300 rounded-md"
        />
      </div>
      <div
        class="absolute inset-0 z-10 flex items-center justify-center gap-2 mx-3 my-5 bg-white opacity-0 bg-opacity-95 group-hover:opacity-100"
      >
        <div
          class="w-12 h-12 bg-gray-300 rounded-full"
        />
        <div
          class="w-12 h-12 bg-gray-300 rounded-full"
        />
      </div>
      <div
        class="flex flex-col items-center justify-between h-full px-4 py-4 bg-white"
      >
        <div
          class="w-full h-6 bg-gray-300 rounded-md mb-1"
        />
        <div
          class="w-1/3 h-4 bg-gray-300 rounded-md mb-2"
        />
        <div
          class="flex items-center gap-2 w-full justify-center"
        >
          <div
            class="w-1/3 h-5 bg-gray-300 rounded-md"
          />
          <div
            class="w-1/4 h-4 bg-gray-200 rounded-md"
          />
          <div
            class="w-1/4 h-4 bg-gray-200 rounded-md"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
